### ISSUE TITLE
daemon stops working after file edit

### ISSUE THREAD

**Issue by xhrnca00** [microsoft/vscode-mypy#211]
When running with `"mypy-type-checker.preferDaemon": true`, the daemon returns the error output only once. All following updates to the file result in no output (even in the logs). This is weird because the normal no-daemon has no such issue. I tried both the bundled and runtime import strategies, Python 3.9 and 3.11.

To reproduce:
0. Create a new profile with just the Mypy Type Checker extension (also installs Python and Pylance).
1. Set `"mypy-type-checker.preferDaemon": true`
2. Create a new file with the following content:
  ```python
  def func(a: float, b: str) -> float:
      return a + b

  func(1, "3")
```
4. Open the file in VSCode - you should see an `Unsupported operand types for + ("float" and "str")` error.
5. Edit the type of `b` from `str` to `float`
6. Save the file - the error will disappear and **no other error should appear**
7. Restart the Mypy server - a new error will appear (there are mismatched types in the function call on line 4)

Expected behavior: the new error should appear immediately after saving without having to reload the extension

Extension log (from doing the steps above):
[log.txt](https://github.com/microsoft/vscode-mypy/files/13257911/log.txt)

**Comment by karthiknadig** [comment]
@xhrnca00 This is a issue with `mypy` daemon itself. Turn it off for your scenarios.
related https://github.com/python/mypy/issues/15677
